validator_client/src/validator_store.rs

```diff
@@ -171,30 +260,521 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
         validator_committee_position: usize,
         attestation: &mut Attestation<E>,
     ) -> Option<()> {
-        // TODO: check for slashing.
-        self.validators
-            .read()
-            .get(validator_pubkey)
-            .and_then(|validator_dir| {
-                let voting_keypair = validator_dir.voting_keypair.as_ref()?;
-
-                attestation
-                    .sign(
-                        &voting_keypair.sk,
-                        validator_committee_position,
-                        &self.fork()?,
-                        &self.spec,
+        let validators = self.validators.read();
+
+        // Retrieving the corresponding ValidatorDir
+        let validator = match validators.get(validator_pubkey) {
+            Some(validator) => validator,
+            None => return None,
+        };
+
+        if validator.attestation_slashing_protection.is_none() {
+            error!(
+                self.log,
+                "Validator does not have attestation slashing protection";
+                "action" => "refused to produce attestation",
+                "pubkey" => format!("{:?}", &validator.voting_keypair.pk),
+            )
+        }
+
+        // Checking for slashing conditions
+        let is_slashing_free = validator
+            .attestation_slashing_protection
+            .as_ref()?
+            .try_lock()?
```

Comment:
What should we do here if it fail? Need to check upstream which behaviour we want?